packages/ckeditor5-widget/src/widget.ts

```diff
@@ -247,6 +244,21 @@ export default class Widget extends Plugin {
 		const modelElement = editor.editing.mapper.toModelElement( element );
 
 		this._setSelectionOverElement( modelElement! );
+
+		function setSelectionInAncestorElement() {
+			if ( element ) {
+				const mapper = editor.editing.mapper;
+				const viewElement = element.is( 'attributeElement' ) ?
+					element.findAncestor( element => !element.is( 'attributeElement' ) )! : element;
+				const modelElement = mapper.toModelElement( viewElement )!;
+
+				domEventData.preventDefault();
+
+				editor.model.change( writer => {
+					writer.setSelection( modelElement, 'in' );
+				} );
+			}
```

Comment:
Two tiny formatting suggestions.

1. To reduce unnecessary nesting, I like to use the [early return pattern](https://gomakethings.com/the-early-return-pattern-in-javascript/). In this case, we can reduce the nesting by one level by returning early if `element` is falsy.

```js
if ( !element ) {
  return;
}

// The rest of the code
```

2. Conditional (ternary) operator is a little bit more readable when true and false expressions are in separate lines:

```js
const viewElement = element.is( 'attributeElement' ) ?
  element.findAncestor( element => !element.is( 'attributeElement' ) )! :
  element;
```